dev-python/pytest-qt/pytest-qt-3.3.0.ebuild

```diff
@@ -0,0 +1,23 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit distutils-r1
```

Comment:
I couldn't run tests without installing this package first, because it calls for its own modules while testing. So you'll have to somehow append PYTHON_LIBDIR to current BUILD_DIR. 

It spawns X windows while testing, so you should use *virtualx.eclass* here. Luckily the docs are good on how to use that.
https://devmanual.gentoo.org/eclass-reference/virtualx.eclass/index.html

Reply:
I added virtx.eclass, and 
```
python_test() {
        virtx pytest -vv || die "Tests fail with ${EPYTHON}"
}
```
and
```
BUILD_DIR+="${PYTHON_LIBDIR}"
```
But I don't think it's working:
```
____________________________________________________________________________ ERROR at setup of test_args ____________________________________________________________________________'
E           and: 'file /var/tmp/portage/dev-python/pytest-qt-3.3.0/temp/pytest-of-portage/pytest-0/test_qapp_args0/test_qapp_args.py, line 1'
E           and: '  def test_args(qapp):'
E           and: "E       fixture 'qapp' not found"
```
What should I do?

Reply:
I haven't studied upstream build system or local build dir, but you'll want to include your current TEMPORARY BUILD_DIR for current interpreter into some of these, 
https://wiki.gentoo.org/wiki/Project:Python/python-utils-r1#PYTHON_SITEDIR
it's most likely SITEDIR with distutils that's needed.

Reply:
I think I got it:
```
PYTHON_SITEDIR+="${WORKDIR}/${PN}-${EPYTHON}/lib/"
```
There now is only one failure: 
```
pytestqt.exceptions.TimeoutError: widget <PyQt5.QtWidgets.QWidget object at 0x7f154e90a050> not activated in 1000 ms.
tests/test_basics.py:103: TimeoutError
```
Not sure if we can fix that, I checked that the module PyQt5.QtWidgets is present (and it is), looks like it is just timing out for some strange reason.

Reply:
Sounds like something to do with *virtualx.eclass*. Hmm...

Reply:
> Sounds like something to do with _virtualx.eclass_. Hmm...

Yeah probably, I just tested `pytest -vv` on the source code as regular user and it works just fine after exporting `PYTHONPATH` to include the source files. I did notice that right at the beginning of the tests a window thingy very briefly appears. Perhaps this is what fails, as this does not happen when running pytest through emerge (probably because emerge doesn't allow pytest to show windows). Should we disable that specific test? What do you think?

Reply:
I just added a patch here that removes the problematic test.

I also used a similar fix as you suggested (with `virtx` and `PYTHON_SITEDIR`) to fix the tests for qtawesome.

Reply:
So it turns out that any environmental variables set in the ebuild are not set inside `virtx`. So when I added `virtx` I broke the tests again when the package is not already installed.

I also discovered that emerge or distutils-r1.eclass or something automatically sets `PYTHONPATH` to `${BUILDDIR}/lib`, this was what was causing the problem in the first place. For this package it should be `${WORKDIR}/${P}` for tests to execute correctly. It should be working correctly now, please let me know if it works for you to.